crud/src/main/java/com/redhat/lightblue/crud/mongo/Translator.java

```diff
@@ -693,45 +713,154 @@ private String writeJSForLoop(StringBuilder bld, Path p, String varPrefix) {
         return arr.toString();
     }
 
-    private DBObject translateFieldComparison(FieldComparisonExpression expr) {
-        StringBuilder str = new StringBuilder(128);
+    private static final String ARR_ARR_EQ=
+        "if(this.%s.length==this.%s.length) { "+
+        "  var allEq=true;"+
+        "  for(var i=0i<this.%s.length;i++) { "+
+        "     if(this.%s[i] != this.%s[i]) { allEq=false; break; } "+
+        "  } "+
+        "  if(allEq) return true;"+
+        "}";
+
+    private static final String ARR_ARR_NEQ=
+        "if(this.%s.length==this.%s.length) { "+
+        "  var allEq=true;"+
+        "  for(var i=0i<this.%s.length;i++) { "+
+        "     if(this.%s[i] != this.%s[i]) { allEq=false; break; } "+
+        "  } "+
+        "  if(!allEq) return true;"+
+        "} else { return true; }";
+
+    private static final String ARR_ARR_LT=
```

Comment:
One missing paren in the end if, but could be optimized by pulling the condition up.  Likely similar for GT, LTE, and GTE.

```
    "if(this.%s.length <= this.%s.length) {"+
        "var n=this.%s.length<this.%s.length?this.%s.length:this.%s.length;"+
        "var allOk=true;"+
        "for(var i=0;i<n;i++) {"+
        "  if(this.%s[i] >= this.%s[i]) { allOk=false;break;}"+
        "}"+
        "if(allOk) return true;"+
    "}";
```